.github/workflows/r.yml

```
@@ -52,6 +52,71 @@ env:
   DOCKER_VOLUME_PREFIX: ".docker/"
 
 jobs:
+  ubuntu-cpp-versions:
+    name: Arrow C++ Versions (${{ matrix.config.cpp_version }})
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        config:
+        - { cpp_version: "15.0.0" }
+        - { cpp_version: "14.0.2" }
+        - { cpp_version: "13.0.0" }
```

I see your R code uses arithmetic 👌  but since this hardcodes the versions, I wonder if we have an automation of some sort that would minimize the chance this list gets out of date without someone noticing.

---

I'll look into that and can put it up in another PR.

---

This will also work and is a bit more compact. 
```suggestion
        cpp_version: [ "15.0.0" , "14.0.2",  "13.0.0"  ]
```
We can also add updating this to the post release scripts. Do we really want to test against all release versions since minimum version? That will ballon pretty quickly... 

Is it possible to have an incompatibility of say R 16.0.0 with libarrow 15.0.0 that doesn't happen with libarrow 13.0.0 (given that all versions/commits in between have been tested?  If so maybe testing against last release and minimum version would be enough? If not just testing  against minimum version should be sufficent?

---

Ah sorry I missed the comment in check-versions.R. The last 3 versions seems like a decent support window and no :balloon: ing :D We should note this in the docs as well.